xcm/src/v0/multi_asset.rs

```diff
@@ -273,15 +280,14 @@ impl MultiAsset {
 			AllConcreteNonFungible { class } => inner.is_concrete_non_fungible(class),
 			AllAbstractNonFungible { class } => inner.is_abstract_non_fungible(class),
 
+			// TODO: should it not be `if i == id && amount >= a`? for self to contain `inner`, self should contain
+			// larger quantities than inner.
```

Comment:
@gavofyork am I wrong?

Reply:
no - that's correct. i'd hope there's a test which picks that up.

Reply:
there are tests for this now, but I aligned them with the wrong logic. will fix both code and tests.